eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -1300,63 +1301,74 @@ class Channel(val nodeParams: NodeParams, wallet: EclairWallet, remoteNodeId: Pu
       val fundingLocked = FundingLocked(d.commitments.channelId, nextPerCommitmentPoint)
       goto(WAIT_FOR_FUNDING_LOCKED) sending fundingLocked
 
-    case Event(channelReestablish@ChannelReestablish(_, _, nextRemoteRevocationNumber, Some(yourLastPerCommitmentSecret), _), d: DATA_NORMAL)
-      if d.commitments.localCommit.index < nextRemoteRevocationNumber =>
-      // if next_remote_revocation_number is greater than our local commitment index, it means that either we are using an outdated commitment, or they are lying
-      // but first we need to make sure that the last per_commitment_secret that they claim to have received from us is correct for that next_remote_revocation_number minus 1
-      if (keyManager.commitmentSecret(d.commitments.localParams.channelKeyPath, nextRemoteRevocationNumber - 1) == yourLastPerCommitmentSecret) {
-        log.warning(s"counterparty proved that we have an outdated (revoked) local commitment!!! ourCommitmentNumber=${d.commitments.localCommit.index} theirCommitmentNumber=${nextRemoteRevocationNumber}")
-        // their data checks out, we indeed seem to be using an old revoked commitment, and must absolutely *NOT* publish it, because that would be a cheating attempt and they
-        // would punish us by taking all the funds in the channel
-        val exc = PleasePublishYourCommitment(d.channelId)
-        val error = Error(d.channelId, exc.getMessage.getBytes)
-        goto(WAIT_FOR_REMOTE_PUBLISH_FUTURE_COMMITMENT) using store(DATA_WAIT_FOR_REMOTE_PUBLISH_FUTURE_COMMITMENT(d.commitments, channelReestablish)) sending error
-      } else {
-        // they lied! the last per_commitment_secret they claimed to have received from us is invalid
-        throw CommitmentSyncError(d.channelId)
-      }
-
     case Event(channelReestablish: ChannelReestablish, d: DATA_NORMAL) =>
-      if (channelReestablish.nextLocalCommitmentNumber == 1 && d.commitments.localCommit.index == 0) {
-        // If next_local_commitment_number is 1 in both the channel_reestablish it sent and received, then the node MUST retransmit funding_locked, otherwise it MUST NOT
-        log.debug(s"re-sending fundingLocked")
-        val nextPerCommitmentPoint = keyManager.commitmentPoint(d.commitments.localParams.channelKeyPath, 1)
-        val fundingLocked = FundingLocked(d.commitments.channelId, nextPerCommitmentPoint)
-        forwarder ! fundingLocked
-      }
+      channelReestablish match {
+        case ChannelReestablish(_, _, nextRemoteRevocationNumber, Some(yourLastPerCommitmentSecret), _) if Helpers.ourNextLocalCommitmentNumberCheck(d, nextRemoteRevocationNumber) =>
```

Comment:
For some reason I actually inverted the result of the function. Both `ourNextLocalCommitmentNumberCheck` and `theirNextLocalCommitmentNumberCheck` should return `true` when parties are in sync, `false` otherwise. Would that be better?

How about renaming:
- `ourNextLocalCommitmentNumberCheck` -> `checkLocalCommit`
- `theirNextLocalCommitmentNumberCheck` -> `checkRemoteCommit`

Reply:
Ok it makes sense. Returning `true` instead and renaming both functions would be fine.